app/Platform/Actions/ComputeGameSearchTitlesAction.php

```diff
@@ -0,0 +1,387 @@
+     */
+    private function convertRomanNumeralsToNumbers(string $title): string
+    {
+        // Need to reverse the order for regex to match longer numerals first (XX before X).
+        $romanNumerals = array_reverse(self::ROMAN_NUMERAL_MAP, true);
+
+        // Only replace Roman numerals that are complete words.
+        return preg_replace_callback(
+            '/\b(' . implode('|', array_keys($romanNumerals)) . ')\b/i',
+            function ($matches) {
+                $romanNumeral = strtoupper($matches[1]);
+
+                return self::ROMAN_NUMERAL_MAP[$romanNumeral] ?? $romanNumeral;
+            },
+            $title
+        );
+    }
+
+    /**
+     * Convert numbers in titles to their Roman numeral equivalents.
+     *
+     * This allows users to search for "Final Fantasy VII" and find "Final Fantasy 7".
+     * Limited to numbers 1-20 which covers most common game series.
+     */
+    private function convertNumbersToRomanNumerals(string $title): string
+    {
+        // Flip the map to go from numbers to Roman numerals (eg: '7' -> 'VII').
+        $numberToRomanMap = array_flip(self::ROMAN_NUMERAL_MAP);
+
+        return preg_replace_callback(
+            '/\b(' . implode('|', array_keys($numberToRomanMap)) . ')\b/',
+            function ($matches) use ($numberToRomanMap) {
+                return $numberToRomanMap[$matches[1]] ?? $matches[1];
+            },
+            $title
+        );
+    }
+
+    /**
+     * Generate common abbreviations for game titles.
+     *
+     * Creates shortened versions like "FF7" for "Final Fantasy VII" or "SM64" for
+     * "Super Mario 64". This handles the common way users search for games using
+     * series abbreviations combined with numbers.
+     */
+    private function generateAbbreviations(string $title): array
+    {
+        $abbreviations = [];
+
+        // Map of series names to their common abbreviations.
+        // It doesn't matter if some of these overlap, and these aren't case-sensitive.
+        // This is just a way we can influence how heavily Meilisearch weighs certain terms.
+        $seriesAbbreviationMap = [
+            'Call of Duty' => ['CoD'],
+            'Castlevania' => ['CV'],
+            'Devil May Cry' => ['DMC'],
+            'Donkey Kong' => ['DK'],
+            'Dragon Quest' => ['DQ'],
+            'Dragon Warrior' => ['DW'],
+            'F-Zero' => ['FZ'],
+            'Final Fantasy' => ['FF'],
+            'God of War' => ['GoW'],
+            'Grand Theft Auto' => ['GTA'],
+            'Kingdom Hearts' => ['KH'],
+            'Legend of Zelda' => ['Zelda', 'LoZ'],
+            'Mega Man' => ['MM'],
+            'Metal Gear' => ['MG'],
+            'Mortal Kombat' => ['MK'],
+            'Need for Speed' => ['NFS'],
+            'Resident Evil' => ['RE'],
+            'Silent Hill' => ['SH'],
+            'Star Fox' => ['SF'],
+            'Street Fighter' => ['SF'],
```

Comment:
How does it determine which `SF` to prioritize?

Reply:
Neither is prioritized. The purpose isn't to disambiguate between series. We just want to ensure that comon abbreviations are included for a game in the search index. When someone searches for "SF", Meilisearch will return results for both Star Fox and Street Fighter games. Other ranking factors are used to actually order the results (like popularity, etc).

With this in mind though, "SF" is probably a bit too ambitious for Star Fox. Removed in latest.